pkg/query-service/app/cloudintegrations/controller.go

```diff
@@ -433,37 +429,43 @@ func (c *Controller) GetServiceDetails(
 		}
 
 		if config != nil {
-			service.Config = config
+			details.Config = config
 
 			enabled := false
 			if config.Metrics != nil && config.Metrics.Enabled {
 				enabled = true
 			}
 
 			// add links to service dashboards, making them clickable.
-			for i, d := range service.Assets.Dashboards {
+			for i, d := range definition.Assets.Dashboards {
 				dashboardUuid := c.dashboardUuid(
 					cloudProvider, serviceId, d.Id,
 				)
 				if enabled {
-					service.Assets.Dashboards[i].Url = fmt.Sprintf(
-						"/dashboard/%s", dashboardUuid,
-					)
-				} else {
-					service.Assets.Dashboards[i].Url = ""
```

Comment:
the else part was added because, once enabled and if we disabe it back the dashboard was still rendered that's why we need to set it to empty.

Reply:
Makes sense, added